### ISSUE TITLE
Not doesn't support empty value

### ISSUE THREAD

**Issue by xxchan** [sagiegurari/duckscript#326]
### Describe The Bug

```
if not ${a} and ${b}
  ...
else
  ...
end
```
gets error
```
Error while running duckscript: Source: Unknown Line: 36 - Unexpected 'and'
```

### Error Stack

```console
The error stack trace
```

### Code Sample

```rust
/// paste code here
```

**Comment by xxchan** [comment]
Besides,
```
not false and false
```
gives `true` is a little confusing... Although I can kind of understand it.

**Comment by xxchan** [comment]
Hmmm, but why

```
# Unexpected value: false
if true and ( not false )

# ok
if false and ( not false )
```

**Comment by xxchan** [comment]
It seems `eval_condition_for_slice` only support literals..

**Comment by sagiegurari** [comment]
sorry for the confusion, but conditions like in 'if' command are build either from
1. variable
2. command
3. condition

in your case, i figure you think you have a condition, but a condition is defined by "A condition statement is made up of values, or/and keywords and '('/')' groups."
see docs at: 
https://github.com/sagiegurari/duckscript/blob/master/docs/sdk.md#std__flowcontrol__If

the 'not' is not a condition keyword, its also a command:
https://github.com/sagiegurari/duckscript/blob/master/docs/sdk.md#std__Not

which i guess where the confusion is coming from.
so when you have a condition phrase, you can't use embedded commands inside since i have no idea where that command args stop and the condition phrase continues.

so you need to do

```sh
a = get_env ...
b = get_env ...
not_a = not ${a}

if ${not_a} and ${b}
  ...
else
  ...
end
```

**Comment by xxchan** [comment]
Yes, thanks for your answer and I’ve already fixed my problem in this way.

I don’t have strong opinion, and haven’t learned much about the design decisions of duckscript. Do you think make “not” a condition keyword can be more user-friendly?

**Comment by sagiegurari** [comment]
conditions are not part of duckscript. they are commands on their own. 
it has its limitations but also extendibility where the main goal was to be super super simple lang with nothing in it just simple 1 liner syntax.
so i agree it can make things simpler, it is also not that clear where not ends and make things more complex at same time.
